.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  # Also allow running this workflow manually from the Actions tab.
+  workflow_dispatch:
+
+jobs:
+  unit-tests:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
```

Comment:
For some reason, GitHub says they offer [more cores for MacOS virtual machines](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources). It is unimportant and does not matter now, but we might revisit it later if we decide it would speed things up or help us detect parallelism bugs

Reply:
More cores, yes, but there is more contention for MacOS runners (we get allocated fewer), so when there are a number of jobs running concurrently, progress tends to stall on our MacOS jobs. Also, Linux usually runs faster to begin with. My guess is that they're offering the extra cores to compensate for the poor performance...